airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java

```diff
@@ -857,6 +867,168 @@ public void testSyncAfterUpgradeToPerStreamStateWithNoNewData(final TestInfo tes
     testHarness.assertSourceAndDestinationDbInSync(WITHOUT_SCD_TABLE);
   }
 
+  @Test
+  public void testResetAllWhenSchemaIsModified() throws Exception {
+    final String sourceTable1 = "test_table1";
+    final String sourceTable2 = "test_table2";
+    final String sourceTable3 = "test_table3";
+    final String outputPrefix = "output_namespace_public.output_table_";
+    final Database sourceDb = testHarness.getSourceDatabase();
+    final Database destDb = testHarness.getDestinationDatabase();
+    sourceDb.query(ctx -> {
+      ctx.createTableIfNotExists(sourceTable1).columns(DSL.field("name", SQLDataType.VARCHAR)).execute();
+      ctx.truncate(sourceTable1).execute();
+      ctx.insertInto(DSL.table(sourceTable1)).columns(DSL.field("name")).values("john").execute();
+      ctx.insertInto(DSL.table(sourceTable1)).columns(DSL.field("name")).values("bob").execute();
+
+      ctx.createTableIfNotExists(sourceTable2).columns(DSL.field("value", SQLDataType.VARCHAR)).execute();
+      ctx.truncate(sourceTable2).execute();
+      ctx.insertInto(DSL.table(sourceTable2)).columns(DSL.field("value")).values("v1").execute();
+      ctx.insertInto(DSL.table(sourceTable2)).columns(DSL.field("value")).values("v2").execute();
+      return null;
+    });
+
+    final UUID sourceId = testHarness.createPostgresSource().getSourceId();
+    final AirbyteCatalog catalog = testHarness.discoverSourceSchema(sourceId);
+    final UUID destinationId = testHarness.createDestination().getDestinationId();
+    final OperationRead operation = testHarness.createOperation();
+    final String name = "test_reset_when_schema_is_modified_" + UUID.randomUUID();
+
+    LOGGER.info("Discovered catalog: {}", catalog);
+
+    final ConnectionRead connection =
+        testHarness.createConnection(name, sourceId, destinationId, List.of(operation.getOperationId()), catalog, null);
+    LOGGER.info("Created Connection: {}", connection);
+
+    sourceDb.query(ctx -> {
+      prettyPrintTables(ctx, sourceTable1, sourceTable2);
+      return null;
+    });
+
+    // Run initial sync
+    LOGGER.info("Running initial sync");
+    final JobInfoRead syncRead =
+        apiClient.getConnectionApi().syncConnection(new ConnectionIdRequestBody().connectionId(connection.getConnectionId()));
+    waitForSuccessfulJob(apiClient.getJobsApi(), syncRead.getJob());
+
+    // Some inspection for debug
+    destDb.query(ctx -> {
+      prettyPrintTables(ctx, outputPrefix + sourceTable1, outputPrefix + sourceTable2);
+      return null;
+    });
+    final ConnectionState initSyncState =
+        apiClient.getConnectionApi().getState(new ConnectionIdRequestBody().connectionId(connection.getConnectionId()));
+    LOGGER.info("ConnectionState after the initial sync: " + initSyncState.toString());
+
+    testHarness.assertSourceAndDestinationDbInSync(false);
```

Comment:
nit: could you use the `WITH_SCD_TABLE` and `WITHOUT_SCD_TABLE` constants in the calls to this method, to improve the readability a bit?